pkg/epp/flowcontrol/framework/policies.go

```diff
@@ -0,0 +1,69 @@
+/*
+Copyright 2025 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package framework
+
+import "sigs.k8s.io/gateway-api-inference-extension/pkg/epp/flowcontrol/types"
+
+// ItemComparatorFunc defines the function signature for comparing two `types.QueueItemAccessor` instances to determine
+// their relative dispatch priority.
+//
+// An implementation of this function determines if item 'a' should be dispatched before item 'b'. It returns true if
+// 'a' is of higher priority, and false otherwise. The specific criteria for "higher priority" (e.g., earlier deadline,
+// lower enqueue time) are defined by the `IntraFlowDispatchPolicy` that vends this function via an `ItemComparator`.
+type ItemComparatorFunc func(a, b types.QueueItemAccessor) bool
+
+// ItemComparator encapsulates the logic for comparing two `types.QueueItemAccessor` instances to determine their
+// relative dispatch priority. It is the definitive source of ordering truth for a flow's queue.
+//
+// It is vended by an `IntraFlowDispatchPolicy` and used by `SafeQueue` implementations that support the
```

Comment:
@kfswain IntraFlowDispatchPolicy will be introduced in the next PR. It vends the `ItemComparator` used to configure the `SafeQueue`, so I had to introduce the type alongside the queues before the full context is present in the repo.